### PR TITLE
pinot: persistent volume

### PR DESCRIPTION
**Pull request by kbastani**
This commit adds the ability to run Pinot with a persistent volume that uses an external Zookeeper dependency that is now included in the formula. This change was necessary as of 0.10.0 to allow users to persist their database state after a restart.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/pinot.rb
Comment (SMillerDev) [inline Formula/pinot.rb:28]: Why this particular port?

Comment (kbastani) [reply Formula/pinot.rb:28]: This is the default port of Zookeeper, which has been added as a dependency.

Comment (SMillerDev) [reply Formula/pinot.rb:28]: Did pinot not work before?

Comment (kbastani) [reply Formula/pinot.rb:28]: The confusion I think here for anyone would be that the `QuickStart` command never used to be the way you would run a production deployment of Pinot (as of `0.9.3`). It was more like a playground for a user to evaluate the database locally. But as of `0.10.0` the `DEFAULT` quickstart recipe can be used for integration testing, evaluation, or even single node production deployments.

Comment (SMillerDev) [reply Formula/pinot.rb]: Homebrew generally doesn't aim to be production ready though, we ship the easiest configuration for developers.

**Comment by kbastani** [comment]
This is the easiest version of configuration. Please merge this simple
change or we will ask for our platform to be removed from Hombrew.

Thanks,

Kenny

**Comment by SMillerDev** [comment]
> This is the easiest version of configuration. 

But I'm wondering if it's the best for Homebrew users, who have come to expect the existing behaviour.

> Please merge this simple change or we will ask for our platform to be removed from Homebrew.

If you pull the open source license we'll happily comply with that.

**Comment by kbastani** [comment]
With alll do respect, you do not understand the work that has gone into a
this. This is an OLAP data store born at LinkedIn/Facebook/Microsoft/Uber.
We and those organizations will collectively shun  Hombrew because this
commit is rejected. For whatever your reasons may be, i ensure you that no
OSS project under those companies will continue to support HB.

**Comment by kbastani** [comment]
> With alll do respect, you do not understand the work that has gone into a this. This is an OLAP data store born at LinkedIn/Facebook/Microsoft/Uber. We and those organizations will collectively shun Hombrew because this commit is rejected. For whatever your reasons may be, i assure you that no OSS project under those companies will continue to support HB.
> […](#)

We need to move forward on this. Either from a customer perspective or here we pay a subscription fee or that this PR is denied on bad merits. I’m open to a phone call. But let’s solve this so all parties win.

**Review by Bo98 — APPROVED** [review]
Seems fine to me. The 0.10.0 changes you've explained was useful for understanding the rationale behind this.

**Comment by Bo98** [comment]
> With alll do respect, you do not understand the work that has gone into a
this. This is an OLAP data store born at LinkedIn/Facebook/Microsoft/Uber.
We and those organizations will collectively shun  Hombrew because this
commit is rejected. For whatever your reasons may be, i assure you that no
OSS project under those companies will continue to support HB.

This is however a very poor and unnecessary tone to direct towards volunteer open source maintainers and I will not be engaging in this PR any further beyond my provisional approval.

**Comment by kbastani** [comment]
This was poor form by me. Frustrations can run high and my tone was not
patient and caring enough not to overreact. I will be better in the future.
Thank you, Bo. I do hope you do involve yourself in providing future
insights into the project and this formula. I am very appreciative  and
thankful for your comments.

**Review by carlocab — APPROVED** [review]
LGTM

**Comment by MikeMcQuaid** [comment]
> This is the easiest version of configuration. Please merge this simple change or we will ask for our platform to be removed from Hombrew.

You can ask and we will say no.

> This is an OLAP data store born at LinkedIn/Facebook/Microsoft/Uber. We and those organizations will collectively shun Hombrew because this commit is rejected. For whatever your reasons may be, i assure you that no OSS project under those companies will continue to support HB.

From your LinkedIn: it seems you work at StarTree and not LinkedIn/Facebook/Microsoft/Uber. I work for GitHub that is owned by Microsoft. I can state categorically that what you have stated here is a lie.

These sort of threats are incredibly unkind and unprofessional. 

> I will be better in the future.

Yes, you will. Even any maintainer sees even the slightest inkling of behaviour trending in this direction: you will be blocked from this project forever.

---

I would like to hold off merging this for 24 hours to make a point here that making these sorts of threats is not something we're going to tolerate here. If no maintainers have an objection in 24 hours: I'm fine with this being merged.

**Comment by kbastani** [comment]
> > This is the easiest version of configuration. Please merge this simple change or we will ask for our platform to be removed from Hombrew.
> 
> You can ask and we will say no.
> 
> > This is an OLAP data store born at LinkedIn/Facebook/Microsoft/Uber. We and those organizations will collectively shun Hombrew because this commit is rejected. For whatever your reasons may be, i assure you that no OSS project under those companies will continue to support HB.
> 
> From your LinkedIn: it seems you work at StarTree and not LinkedIn/Facebook/Microsoft/Uber. I work for GitHub that is owned by Microsoft. I can state categorically that what you have stated here is a lie.
> 
> These sort of threats are incredibly unkind and unprofessional.
> 
> > I will be better in the future.
> 
> Yes, you will. Even any maintainer sees even the slightest inkling of behaviour trending in this direction: you will be blocked from this project forever.
> 
> I would like to hold off merging this for 24 hours to make a point here that making these sorts of threats is not something we're going to tolerate here. If no maintainers have an objection in 24 hours: I'm fine with this being merged.

This is a fair response and action. But in your reaction here, nothing I said was a lie. The open source project that is maintained by StarTree was founded and is now maintained by those companies. StarTree was spun out of LinkedIn where the project incubated. Please don’t treat me that way. I am a user and a contributor and a maintainer should try to forge bonds and find mutual ground with the value that others add to it. I appreciate your work, and apologize for my overreaction. Please come my way with that empathy and we could help each other rather than sanction threats.

**Comment by MikeMcQuaid** [comment]
> But in your reaction here, nothing I said was a lie.

Saying Microsoft will shun Homebrew if your commit was rejected was and is a lie.

---

Unfortunately this has escalated off GitHub so we're done here.

If anyone else wishes to submit this PR without shouting about how terrible Homebrew and our maintainers are: please feel free.

**Pull request closed without merging**
